### ISSUE TITLE
Agora node does not catch up after restart

### ISSUE THREAD

**Issue by kchulj** [bosagora/agora#1512]
Currently, an agora does not catch up after restart.
It should get the latest blocks and catch up with the blockchain.

**Comment by kchulj** [comment]
It was observed that the restarted node tries to enroll, but gets rejected.
```
2021-01-20 10:37:01,233 Info [agora.consensus.state.ValidatorSet] - Rejected pre-image: validator not enrolled for key: 0x790ab7...
```
This might happen because the `utxo_key` of the restarted node is different from the the `utxo_key` in the genesis block enrollment. It somehow got modified. The `utxo_key` (`enroll_key`) of the other nodes match the ones in the genesis.

**Comment by omerfirmak** [comment]
>It was observed that the restarted node tries to enroll, but gets rejected.

It's expected, if the network is at Height 1999.

**Comment by kchulj** [comment]
Currently it is not up to 1999 yet, and the other nodes are able to re-enroll just fine. The restarted one is strange because it can't.

**Comment by omerfirmak** [comment]
> Currently it is not up to 1999 yet, and the other nodes are able to re-enroll just fine. The restarted one is strange because it can't.

If the restarted node has not caught up to the network yet, that would explain it as well. It would send bunch of invalid enrollment requests as it is discovering new blocks and at the `start()`

**Comment by kchulj** [comment]
> If the restarted node has not caught up to the network yet, that would explain it as well. It would send bunch of invalid enrollment requests as it is discovering new blocks and at the `start()`

The node is not attempting to catch up to the network or discover.. That may be the problem.

**Comment by omerfirmak** [comment]
Well, agora says that is failing to listen to the port 9110. But netstat says the opposite :)

agora:
```
Jan 20 12:23:27 na-001 docker[19876]: Failed to listen on 0.0.0.0:9110
```

netstat:
```
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.11:34279        0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:4826            0.0.0.0:*               LISTEN      1/agora
tcp        0      0 :::9110                 :::*                    LISTEN      1/agora
```

**Comment by ferencdg** [comment]
I see this in agora service config in ansible:

//SIGKILL only manually
TimeoutStopSec=infinity
SendSIGKILL=no
Restart=on-failure
User=agora
Group=nogroup

I know sometimes vibe.d have difficulties terminating and it keeps processes using it hanging... Are you sure the previous agora instances were termintated properly?

**Comment by omerfirmak** [comment]
Well, `top` only shows a single instance. But I am not sure if it's the newly created one.

I even stopped, pruned and restarted the container, but did not help.

**Comment by Geod24** [comment]
Will be fully fixed once #1662 is merged.
